packages/vuetify/src/directives/colorable/index.ts

```diff
@@ -0,0 +1,111 @@
+import { VNode, VNodeDirective } from 'vue'
+import colors from '../../util/colors'
+
+interface BorderModifiers {
+  top?: Boolean
+  right?: Boolean
+  bottom?: Boolean
+  left?: Boolean
+}
+
+function isCssColor (color?: string | false): boolean {
+  return !!color && !!color.match(/^(#|var\(--|(rgb|hsl)a?\()/)
+}
+
+function classToHex (color: string, colors: Record<string, Record<string, string>>): string {
+  const [colorName, colorModifier] = color
+    .toString().trim().replace('-', '').split(' ', 2) as (string | undefined)[]
+
+  let hexColor = ''
+  if (colorName && colorName in colors) {
+    if (colorModifier && colorModifier in colors[colorName]) {
+      hexColor = colors[colorName][colorModifier]
+    } else if ('base' in colors[colorName]) {
+      hexColor = colors[colorName].base
+    }
+  }
+  return hexColor
+}
+
+function setTextColor (el: HTMLElement, color: string) {
+  const setColor = (v: string) => {
+    el.style.color = v
+    el.style.caretColor = v
+  }
+
+  if (isCssColor(color)) setColor(color)
+  else if (color) setColor(classToHex(color, colors))
```

Comment:
imo I would prefer something like this as it reduces the current complexity and gets rid of the extra function. the same can be done for the rest of the functions.
```
const cssColor = !isCssColor(color) ? classToHex(color, colors) : color

el.style.color = cssColor
e.style.caretColor = cssColor
```

Reply:
Ah, I knew something wasn't right when I was writing it, thx